pkg/storage/requests.go

```diff
@@ -32,24 +33,37 @@ func (s *Store) FetchMyRequests(ctx context.Context) ([]Request, error) {
 	requests := []Request{}
 
 	requestsSQL := `
-		SELECT 
-			id,
-			name,
-			created_at AS submitted_at,
-			'ACCESSIBILITY_REQUEST' record_type
-		FROM accessibility_requests
-			WHERE deleted_at IS NULL
-			AND eua_user_id = $1
-		UNION
-		SELECT
-			id,
-			project_name AS name,
-			submitted_at,
-			'GOVERNANCE_REQUEST' record_type
-		FROM system_intakes
-			WHERE archived_at IS NULL
-			AND eua_user_id = $1
-		ORDER BY submitted_at desc nulls first
+	SELECT 
+		accessibility_requests.id,
+		accessibility_requests.name,
+		accessibility_requests.created_at AS submitted_at,
+		'ACCESSIBILITY_REQUEST' record_type,
+		accessibility_request_status_records.status::text
+	FROM accessibility_requests
+		JOIN accessibility_request_status_records
+			ON accessibility_request_status_records.request_id = accessibility_requests.id
+		INNER JOIN (
+			SELECT
+				request_id,
+				max(created_at)
+			FROM
+				accessibility_request_status_records arsr 
+			GROUP BY request_id
+		) current_status 
+			ON current_status.request_id = accessibility_requests.id
```

Comment:
I think we should be able to accomplish what we need without have two joins to the same table here.